src/player.cpp

```diff
@@ -73,6 +73,44 @@ Player::~Player()
 	setEditHouse(nullptr);
 }
 
+bool Player::stopLiveCasting() {
+	liveCasting = false;
+	castPassword = "";
+
+	for(ProtocolSpectator* spectator : getSpectators()) {
+		spectator->disconnect();
+		removeSpectator(spectator);
+	}
+
+	std::ostringstream query;
+	query << "UPDATE `players_online` SET `cast_password` = NULL, `cast_spectators` = '0';";
+	Database::getInstance()->executeQuery(query.str());
+
+	g_game.removeLiveCaster(this);
+
+	return isLiveCasting() == false;
+}
+
+bool Player::startLiveCasting(const std::string& password) {
+	liveCasting = true;
+
+	if(!password.empty()) {
+		castPassword = password;
+	}
+
+	sendChannel(CHANNEL_CAST, "Live Cast", nullptr, nullptr);
+
+	Database* db = Database::getInstance();
+
+	std::ostringstream query;
+	query << "UPDATE `players_online` SET `cast_password` = '" << db->escapeString(password) << "' WHERE `player_id` = '" << getGUID() <<"';";
```

Comment:
yes in !cast (password) get error in console in !cast and connect to spect get  debug and crash server this code its crappy ...